### PR TITLE
MF-1091 - Use channels. as broker prefix

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

* Use "channels" instead of "channel" as subject prefix of the NATS message broker.

### CODE REVIEW

File: broker/nats.go [hunk 1]
```diff
@@ -64,9 +65,9 @@ func (b broker) Publish(_ context.Context, _ string, msg Message) error {
 		return err
 	}
 
-	subject := fmt.Sprintf("%s.%s", prefix, msg.Channel)
+	subject := fmt.Sprintf("%s.%s", chansPrefix, msg.Channel)
 	if msg.Subtopic != "" {
-		subject = fmt.Sprintf("%s.%s", subject, msg.Subtopic)
+		subject = fmt.Sprintf("%s.%s", chansPrefix, msg.Subtopic)
```
Comment (drasko) [inline broker/nats.go]: Looks like we are changing the logic - we used previously defined `subject` (on the line above), now we are using chan prefix, and this looks incorrect - as subtopic is actually added to a topic, not to a prefix.

**Review by drasko — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
This is an API breaking change for all applications that are using data directly from the NATS, and this should be documented for the new release.

**Comment by manuio** [comment]
@drasko docs updated: https://github.com/mainflux/docs/pull/22

**Review by drasko — APPROVED** [review]
LGTM

File: broker/nats.go [hunk 2]
```diff
@@ -28,9 +28,10 @@ type Nats interface {
 }
 
 const (
-	prefix = "channel"
+	chansPrefix = "channels"
+
 	// SubjectAllChannels define the subject to subscribe to all channels messages
-	SubjectAllChannels = "channel.>"
+	SubjectAllChannels = "channels.>"
```
Comment (nmarcetic) [inline broker/nats.go:34]: Can you please describe this PR, the purpose and reason.

**Review by nmarcetic — CHANGES_REQUESTED** [review]

File: broker/nats.go (hunk 2 above)
Comment (drasko) [reply broker/nats.go:34]: @nmarcetic this is more or less evident. `channels.<chan_id>` is to make it consistent with the public API (which is `channels/<chan_id>`, and not `channel/<chan_id>`.

Comment (manuio) [reply broker/nats.go:34]: In case I updated the description of the PR

**Review by dborovcanin — APPROVED** [review]

**Review by nmarcetic — APPROVED** [review]

**Pull request merged**